src/tests/Microsoft.PowerFx.Interpreter.Tests/DisplayNameTests.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Public.Values;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.PowerFx.Interpreter.Tests
+{    
+    public class DisplayNameTests 
+    {
+
+        [Fact]
+        public void CollisionsThrow()
+        {
+            var engine = new RecalcEngine();
+            RecordType r1 = new RecordType()
+                .Add(new NamedFormulaType("Num", FormulaType.Number), "DisplayNum");
+
+            Assert.Throws<NameCollisionException>(() => r1.Add(new NamedFormulaType("DisplayNum", FormulaType.Date), "NoCollision"));
+            Assert.Throws<NameCollisionException>(() => r1.Add(new NamedFormulaType("NoCollision", FormulaType.Date), "DisplayNum"));
+            Assert.Throws<NameCollisionException>(() => r1.Add(new NamedFormulaType("NoCollision", FormulaType.Date), "Num"));
+        }
+
+        [Fact]
+        public void ConvertToDisplayNames()
```

Comment:
another test case:
adding a comment / extra whitespace and verifying they're preserved during conversion. (ie, it's not just a pretty printer walking the AST).